### ISSUE TITLE
Only stored passwords when connecting via RD-Gateway

### ISSUE THREAD

**Issue by MaFreiberger** [FreeRDP/Remmina#922]
Kubuntu 16.04
remmina 1.2.0-rcgit.14 (git n/a)
connecting to Windows PCs (W7, W10, Server 2012R2, Server 2008 R2) via RD-Gateway (SBS 2011 / Server 2008R2)

Till the actual 1.2 version it was only possible to connect via a dedicated ssh-Gateway into our LAN. Now remmina supports RD-Gateway and it seems to work after you got tested different authentication protocolls.

BUT
connecting via RD-Gateway needs to save the password inside remmina. Because of security issues and the company security this is something i really don't like to do. My passwords are stored in a dedicated password manager and they should not be stored somewhere else.

PLEASE
give us the possibility to type the passwords "on the fly" even when using RD-Gateway like it is possible with a direct RDP or with a "over SSH-Server RDP".

Thank you very much.

Really good work. 
##

**Comment by nike38rus** [comment]
+1, we're experiencing the same problem with RD Gateway. 
If password isn't stored, authentification window is popping out, but after I enter my credentials and click OK I get this error: "Authentication to RDP server <servername> failed. Check username, password and domain."

Console log:
```
(remmina:24254): Gtk-WARNING **: gtk_menu_attach_to_widget(): menu already attached to GtkMenuItem
Gtk-Message: GtkDialog mapped without a transient parent. This is discouraged.
[12:59:08:779] [24254:25127] [INFO][com.freerdp.client.common.cmdline] - loading channel rdpdr
[12:59:08:779] [24254:25127] [INFO][com.freerdp.client.common.cmdline] - loading channel rdpsnd
[12:59:08:779] [24254:25127] [INFO][com.freerdp.client.common.cmdline] - loading channel cliprdr
[12:59:08:779] [24254:25127] [INFO][com.freerdp.core.nego] - Detecting if host can be reached locally. - This might take some time.
[12:59:08:779] [24254:25127] [INFO][com.freerdp.core.nego] - To disable auto detection use /gateway-usage-method:direct
[12:59:14:382] [24254:25127] [ERROR][com.winpr.timezone] - Unable to get current timezone rule
[12:59:14:894] [24254:25127] [INFO][com.freerdp.core.nego] - Detecting if host can be reached locally. - This might take some time.
[12:59:14:894] [24254:25127] [INFO][com.freerdp.core.nego] - To disable auto detection use /gateway-usage-method:direct
[12:59:14:900] [24254:25127] [ERROR][com.freerdp.core] - getaddrinfo: Имя или служба не известны
[12:59:14:960] [24254:25127] [ERROR][com.freerdp.core.gateway.rpc] - error! Status Code: 401
[12:59:14:960] [24254:25127] [ERROR][com.freerdp.core.gateway.http] - HTTP/1.1 401 Unauthorized
[12:59:14:960] [24254:25127] [ERROR][com.freerdp.core.gateway.http] - Content-Type: text/plain
[12:59:14:960] [24254:25127] [ERROR][com.freerdp.core.gateway.http] - Server: Microsoft-IIS/8.5
[12:59:14:960] [24254:25127] [ERROR][com.freerdp.core.gateway.http] - WWW-Authenticate: Basic realm="<RD Gateway hostname here>"
[12:59:14:960] [24254:25127] [ERROR][com.freerdp.core.gateway.http] - WWW-Authenticate: Negotiate
[12:59:14:960] [24254:25127] [ERROR][com.freerdp.core.gateway.http] - WWW-Authenticate: NTLM
[12:59:14:960] [24254:25127] [ERROR][com.freerdp.core.gateway.http] - X-Powered-By: ASP.NET
[12:59:14:960] [24254:25127] [ERROR][com.freerdp.core.gateway.http] - Date: Sun, 17 Jul 2016 06:59:07 GMT
[12:59:14:960] [24254:25127] [ERROR][com.freerdp.core.gateway.http] - Content-Length: 13
[12:59:14:960] [24254:25127] [ERROR][com.freerdp.core] - freerdp_set_last_error ERRCONNECT_AUTHENTICATION_FAILED [0x20009]
[12:59:14:960] [24254:25127] [ERROR][com.freerdp.core.gateway.tsg] - tsg_check failure
[12:59:14:961] [24254:25127] [ERROR][com.freerdp.core.nego] - Protocol Security Negotiation Failure
[12:59:14:961] [24254:25127] [ERROR][com.freerdp.core.connection] - Error: protocol security negotiation or connection failure
Gtk-Message: GtkDialog mapped without a transient parent. This is discouraged.
```

Ubuntu 16.04, Remmina 1.2.0-rcgit.14 (git n/a)

**Comment by antenore** [comment]
I mark this as duplicate of #933 because there were a longer discussion over there.
Please join us as we have implemented the feature in a new branch.

**Comment by eSavior** [comment]
I believe this is a separate issue from #933. #933 wants the ability to provide separate credentials for authenticating into the RD Gateway than will be used to authenticate into the destination machine. This issue is (IMHO) asking for an ability to PROMPT for the RD Gateway credentials if they are not provided, similar to the prompt you get for the destination machine credentials if they are not provided. This would be similar to the behavior of the Microsoft RDP client when a gateway server hostname/IP has been specified, but the username and/or the password has been left blank.

I would like to +1 this request. Can we please reopen this issue with the above understanding? I can also provide access to an RD Gateway server if it is needed for testing purposes.

Thanks!

**Comment by prensing** [comment]
+1.

**Comment by antenore** [comment]
@eSavior can you please share with me and Giovanni an rdp gateway?

We will work on it starting from the second week of January. The emails are in the Remmina about widget

**Comment by eSavior** [comment]
Hi, @antenore. Sorry for taking so long to respond...wanted to make sure that I could comply with your request without violating any procedures on my end. I have a test environment set up and, praise Jesus, finally was given the magic incantations to configure my firewalls correctly. Not sure how to send private messages on GitHub, but feel free to contact me via e-mail directly and I'll let you know how you can access it. God Bless!

**Comment by antenore** [comment]
@eSavior sorry, I forgot about this and in the next weeks I won't have time to take a look.

Is the offer still valid? Can you contact me by email (it's in the about widget in remmina)

cc @ic3d thanks

**Comment by eSavior** [comment]
Yes, sir. Definitely still valid. Had to lug all that equipment out of the garage and reconfigure for my current firewall setup. A lot changes in 5 months around here :-)

Check your simbiosi e-mail and we'll go from there!

**Comment by antenore** [comment]
WARNING:
The Remmina project is moving to gitlab.com, in order to keep receiving notification about this issue,
you should have a gitlab account and link it to github.
The migration will happen in the next days.
Thanks for your support and understanding.

cc @antenore @eSavior @MaFreiberger @nike38rus @prensing
